stdlib/2and3/optparse.pyi

```diff
@@ -0,0 +1,226 @@
+# Generated by pytype, with only minor tweaks. Might be incomplete.
+import sys
+from typing import Any, Optional, List, Callable, Tuple, Dict, Iterable, Union, Mapping, IO
+
+# See https://groups.google.com/forum/#!topic/python-ideas/gA1gdj3RZ5g
+if sys.version_info >= (3,):
+    Text = str
+else:
+    Text = Union[str, unicode]
+
+NO_DEFAULT = ...  # type: Tuple[Text, ...]
+SUPPRESS_HELP = ...  # type: Text
+SUPPRESS_USAGE = ...  # type: Text
+
+def check_builtin(option: Option, opt: Any, value: Text) -> Any: ...
+def check_choice(option: Option, opt: Any, value: Text) -> Any: ...
+if sys.version_info >= (3,):
+    def isbasestring(x: Any) -> bool: ...
+
+class OptParseError(Exception):
+    msg = ...  # type: Text
+    def __init__(self, msg: Text) -> None: ...
+
+class BadOptionError(OptParseError):
+    opt_str = ...  # type: Text
+    def __init__(self, opt_str: Text) -> None: ...
+
+class AmbiguousOptionError(BadOptionError):
+    possibilities = ...  # type: Iterable[Text]
+    def __init__(self, opt_str: Text, possibilities: List[Text]) -> None: ...
+
+class OptionError(OptParseError):
+    msg = ...  # type: Text
+    option_id = ...  # type: Text
+    def __init__(self, msg: Text, option: Option) -> None: ...
+
+class OptionConflictError(OptionError): ...
+
+class OptionValueError(OptParseError): ...
+
+
+class HelpFormatter:
+    NO_DEFAULT_VALUE = ...  # type: Text
+    _long_opt_fmt = ...  # type: Text
+    _short_opt_fmt = ...  # type: Text
+    current_indent = ...  # type: int
+    default_tag = ...  # type: Text
+    help_position = ...  # type: Any
+    help_width = ...  # type: Any
+    indent_increment = ...  # type: int
+    level = ...  # type: int
+    max_help_position = ...  # type: int
+    option_strings = ...  # type: Dict[Option, Text]
+    parser = ...  # type: OptionParser
+    short_first = ...  # type: Any
+    width = ...  # type: int
+    def __init__(self, indent_increment: int, max_help_position: int, width: Optional[int], short_first: int) -> None: ...
+    def _format_text(self, text: Text) -> Text: ...
+    def dedent(self) -> None: ...
+    def expand_default(self, option: Option) -> Text: ...
+    def format_description(self, description: Text) -> Text: ...
+    def format_epilog(self, epilog) -> Text: ...
+    def format_heading(self, heading: Any) -> Text: ...
+    def format_option(self, option: OptionParser) -> Text: ...
+    def format_option_strings(self, option: OptionParser) -> Any: ...
+    def format_usage(self, usage: Any) -> Text: ...
```

Comment:
Which means you can subclass HelpFormatter without overriding the "abstract" methods, but you'll get an error once you call the method. I wonder whether type checkers should warn about that.